### PR TITLE
[#506] Removed reflection call.

### PR DESCRIPTION
**Pull request by ArneDeutsch**
Do not inherit and modify (by reflection) the original
java.util.jar.Manifest but instead C&P and modify the code. Adapt
ManifestMergerTest by importing the new classes.

Signed-off-by: Arne Deutsch <Arne.Deutsch@itemis.de>

### CODE REVIEW

**Comment by cdietrich** [comment]
@szarnekow would be nice if you could review

File: org.eclipse.xtext.util/src/org/eclipse/xtext/util/MergeableManifest.java [hunk 1]
```diff
@@ -418,30 +442,1025 @@ public static String mergeIntoCommaSeparatedList(String currentString, Set<Strin
...
+
+				String value = (String) e.getValue();
+				if (value != null) {
+					byte[] vb = value.getBytes("UTF8");
+					value = new String(vb, 0, 0, vb.length);
+				}
+				buffer.append(value);
+
+				buffer.append("\r\n");
+				MergeableManifest.make72Safe(buffer);
```
Comment (szarnekow) [inline org.eclipse.xtext.util/src/org/eclipse/xtext/util/MergeableManifest.java]: Other places call make512Safe - should we adjust this location, too?

File: org.eclipse.xtext.util/src/org/eclipse/xtext/util/MergeableManifest.java [hunk 2]
```diff
@@ -418,30 +442,1025 @@ public static String mergeIntoCommaSeparatedList(String currentString, Set<Strin
...
+
+					String value = (String) e.getValue();
+					if (value != null) {
+						byte[] vb = value.getBytes("UTF8");
+						value = new String(vb, 0, 0, vb.length);
+					}
+					buffer.append(value);
+
+					buffer.append("\r\n");
+					MergeableManifest.make72Safe(buffer);
```
Comment (szarnekow) [inline org.eclipse.xtext.util/src/org/eclipse/xtext/util/MergeableManifest.java]: make512Safe?

File: org.eclipse.xtext.util/src/org/eclipse/xtext/util/MergeableManifest.java [hunk 3]
```diff
@@ -56,6 +70,22 @@
 	@Deprecated
 	private static final String LINEBREAK = Strings.newLine();
 
+	/**
+	 * Adds line breaks to enforce a maximum 72 bytes per line.
+	 */
+	static void make72Safe(StringBuffer line) {
```
Comment (szarnekow) [inline org.eclipse.xtext.util/src/org/eclipse/xtext/util/MergeableManifest.java]: Looks like we had the intention to use make512Safe at most locations. Should we adjust the remaining pieces, too?

File: org.eclipse.xtext.util/src/org/eclipse/xtext/util/MergeableManifest.java [hunk 4]
```diff
@@ -418,30 +442,1025 @@ public static String mergeIntoCommaSeparatedList(String currentString, Set<Strin
...
+			 * {@code Name} object for {@code Specification-Vendor} manifest
+			 * attribute used for package versioning.
+			 */
+			public static final Name SPECIFICATION_VENDOR = new Name("Specification-Vendor");
+
+			/**
+			 * {@code Name} object for {@code Multi-Release} manifest attribute
+			 * that indicates this is a multi-release JAR file.
+			 *
+			 * @since 9
```
Comment (szarnekow) [inline org.eclipse.xtext.util/src/org/eclipse/xtext/util/MergeableManifest.java]: These since tags will confuse the API tooling

Comment (ArneDeutsch) [reply org.eclipse.xtext.util/src/org/eclipse/xtext/util/MergeableManifest.java]: Will delete.

File: org.eclipse.xtext.util/src/org/eclipse/xtext/util/MergeableManifest.java (hunk 3 above)
Comment (ArneDeutsch) [reply org.eclipse.xtext.util/src/org/eclipse/xtext/util/MergeableManifest.java]: Dead code. Will delete.

File: org.eclipse.xtext.util/src/org/eclipse/xtext/util/MergeableManifest.java (hunk 2 above)
Comment (ArneDeutsch) [reply org.eclipse.xtext.util/src/org/eclipse/xtext/util/MergeableManifest.java]: Dead code. Will delete.

File: org.eclipse.xtext.util/src/org/eclipse/xtext/util/MergeableManifest.java (hunk 1 above)
Comment (ArneDeutsch) [reply org.eclipse.xtext.util/src/org/eclipse/xtext/util/MergeableManifest.java]: I have tried to modify existing code as less as possible. Checked again and found that it is dead code. Will delete the dead code.

**Comment by ArneDeutsch** [comment]
Hi @szarnekow . I have fixed your findings (removed dead code). But am not sure if I have messed up the comit together with the version commit. Should I close this pull request and request a new one?

**Comment by ArneDeutsch** [comment]
I opened https://github.com/eclipse/xtext-core/pull/750 in replacement of this one.

**Pull request closed without merging**